### DISCUSSION TITLE
remove @import from css file?

### DISCUSSION THREAD

**Discussion by jf-home** [anantoghosh/gatsby-plugin-purgecss#1006] [Q&A]
Hi, 

A dependency in my Gatsby site has a css file with an @import statement that I want to strip out using gatsby-plugin-purgecss.  Is this possible and, if so, how would I go about achieving it?

The full import statement is as follows: 

`@import url(https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap);`

It's a real pain as the css file isn't needed and, because it's an externally-referenced file, it's hitting the page performance due to blocking page rendering.

Most grateful for any help you can offer!

**Comment by anantoghosh** [comment]
Unfortunately, I don't think this is possible with purgecss for now, you can try to ask around the purgecss repo https://github.com/FullHuman/purgecss.

Instead, I would recommend that you write a webpack loader which removes this line. The loader should be quite straight forward to write, you get the css as input and call `callback` with the modified css. Here's how the loader for this plugin was written https://github.com/anantoghosh/gatsby-plugin-purgecss/blob/master/packages/gatsby-plugin-purgecss/src/loader.ts.

**Reply by jf-home** [reply]
@anantoghosh - thanks for the prompt reply - much appreciated!

I'm happy to learn but this is a bit beyond me at the moment.  Is there any chance that you could break it down a little more, please?  I have been trying all sorts of things with Webpack in gatsby-node and webpack.config.js - all to no avail sadly!

I have spent a fair amount of time searching the web too but nothing has been particularly enlightening so far.  

Thanks in advance!

**Comment by jf-home** [comment]
I managed to get to a workable solution on this - it just took me a little time to piece things together.  

I used the Gatsby Node API - `onPreBootstrap` - within which I added code to read the contents of the css file in the node_modules folder and place it into a variable, then I replaced the unwanted string (`@import...`) in the variable, and then used the same variable to write the modified css back to the same file and location.

From that, the resulting css and html files created by the build process no longer contain the unwanted @import statement, resulting in no more render blocking!

In case anyone has a similar requirement in the future, here is the code that I added to the `gatsby-node.js` file: 

```
exports.onPreBootstrap = async ({ graphql }) => {
    function fromDir(startPath, filter, findString, replaceString) {
      if (!fs.existsSync(startPath)) {
        console.log('no dir ', startPath)
        return
      }

      var files = fs.readdirSync(startPath)
      for (var i = 0; i < files.length; i++) {
        var filename = path.join(startPath, files[i])
        var stat = fs.lstatSync(filename)
        if (stat.isDirectory()) {
          fromDir(filename, filter) //recurse
        } else if (filename.endsWith(filter)) {
          try {
            let data = fs.readFileSync(filename, 'utf8')
            data = data.replace(findString, replaceString)

            try {
              fs.writeFileSync(filename, data)
              // file written successfully
              console.log(`\u001b[32msuccess\u001b[0m ${filename} updated!`);
            } catch (err) {
              console.error(err)
            }
          } catch (err) {
            console.error(err)
          }
        }
      }
    }

    fromDir( 'node_modules/<package-folder-structure-to-the-css-file>/', '.css', '@import url("https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap");', '' )
}
```

Thanks for your help @anantoghosh as it got me exploring options and thinking about different approaches.
